File: src/main/java/co/gatedaccess/web/data/model/Occupant.java
package co.gatedaccess.web.data.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.mongodb.lang.NonNull;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Document
public class Occupant {
    @Id
    String id;
    @Indexed
    @Field("first_name")
    String firstName;
    @Indexed
    @Field("last_name")
    String lastName;
    String phone;
    String gender;
    @NonNull
    @DBRef(lazy = true)
    Member guardian;
    @Field("photo_url")
    String photoUrl;
    @Field("created_at")
    @JsonProperty("created_at")
    @CreatedDate
    Date createdAt;

    public Occupant(String id, String firstName, String lastName, String phone, String gender, @NonNull Member guardian, String photoUrl, Date createdAt) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.phone = phone;
        this.gender = gender;
        this.guardian = guardian;
        this.photoUrl = photoUrl;
        this.createdAt = createdAt;
    }
}

File: src/main/kotlin/co/gatedaccess/web/data/model/Occupant.kt
package co.gatedaccess.web.data.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.mongodb.lang.NonNull
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import java.util.*

@Document
class Occupant(
    @Id var id: String,
    @Field(
        "first_name"
    ) @field:Indexed var firstName: String,
    @Field(
        "last_name"
    ) @Indexed var lastName: String,
    var phone: String,
    var gender: String,
    @DBRef(
        lazy = true
    ) @field:NonNull @param:NonNull var guardian: Member,
    @field:Field(
        "photo_url"
    ) var photoUrl: String,
    @field:CreatedDate @field:JsonProperty("created_at") @field:Field("created_at") var createdAt: Date
)
